cpp/open3d/pipelines/integration/UniformTSDFVolume.cpp

```diff
@@ -138,6 +154,64 @@ std::shared_ptr<geometry::PointCloud> UniformTSDFVolume::ExtractPointCloud() {
     return pointcloud;
 }
 
+std::shared_ptr<geometry::TriangleMesh>
+UniformTSDFVolume::ExtractTriangleMesh_v2() {
+    auto mesh = std::make_shared<geometry::TriangleMesh>();
+    //float *image_array = new float[resolution_  * resolution_ * resolution_];
+    //const long r_sq = resolution_ * resolution_;
+    int dims[] = {resolution_, resolution_, resolution_};
+    float ***values = new float **[dims[0]];
+    for(int i = 0; i <= dims[1]; ++i) {
+        values[i] = new float *[dims[2]];
+        for (int j = 0; j <= dims[1]; ++j) {
+            values[i][j] = new float[dims[2]];
+        }
+    }
+
+    for (int x = 0; x < resolution_ - 1; x++) {
+        for (int y = 0; y < resolution_ - 1; y++) {
+            for (int z = 0; z < resolution_ - 1; z++) {
+                Eigen::Vector3i idx0(x, y, z);
+                const float f = voxels_[IndexOf(idx0)].tsdf_;
+                values[x][y][z] = f;
```

Comment:
Only TSDF is used here, weights are not handled, see the [original implementation](https://github.com/isl-org/Open3D/blob/f8c94a1993d08cfa7ec974a76dea8b61c7304946/cpp/open3d/pipelines/integration/UniformTSDFVolume.cpp#L128). This is known to provide problematic mesh especially around boundaries.